FinSearchUnified/Tests/BusinessLogic/Models/FindologicArticleModelTest.php

```diff
@@ -138,4 +144,108 @@ public function testEmptySuppliersAreSkipped(array $articleConfiguration)
         );
         $this->assertEquals(get_class($findologicArticle), FindologicArticleModel::class);
     }
+
+    /**
+     * @dataProvider articlesWithAttributeProvider
+     *
+     * @param array $articleConfiguration
+     * @param string $expected
+     *
+     * @throws ReflectionException
+     */
+    public function testArticleWithAttributes(array $articleConfiguration, $expected)
+    {
+        $baseCategory = new Category();
+        $baseCategory->setId(5);
+
+        $articleFromConfiguration = $this->createTestProduct($articleConfiguration);
+
+        $findologicArticle = $this->articleFactory->create(
+            $articleFromConfiguration,
+            'ABCD0815',
+            [],
+            [],
+            $baseCategory
+        );
+
+        $xmlArticle = $findologicArticle->getXmlRepresentation();
+
+        $reflector = new ReflectionClass(Item::class);
+        $properties = $reflector->getProperty('properties');
+        $properties->setAccessible(true);
+
+        $properties = $properties->getValue($xmlArticle);
+
+        $propertiesArray = current($properties);
+        $this->assertArrayHasKey('attr1', $propertiesArray);
+        $this->assertSame($expected, $propertiesArray['attr1']);
+    }
+
+    public function articlesWithAttributeProvider()
+    {
+        return [
+            'Article Attribute' => [
+                [
+                    'name' => 'FindologicArticle 1',
+                    'active' => true,
+                    'tax' => 19,
+                    'supplier' => "Findologic",
+                    'categories' => [
+                        ['id' => 3],
+                        ['id' => 5],
+                    ],
+                    'images' => [
+                        ['link' => 'https://via.placeholder.com/300/F00/fff.png'],
+                        ['link' => 'https://via.placeholder.com/300/09f/000.png'],
+                    ],
+                    'mainDetail' => [
+                        'number' => 'FINDOLOGIC1',
+                        'active' => true,
+                        'inStock' => 16,
+                        'prices' => [
+                            [
+                                'customerGroupKey' => 'EK',
+                                'price' => 99.34,
+                            ],
+                        ]
+                    ],
+                    'attribute' => [
+                        'attr1' => 'Article Attribute'
+                    ]
+                ],
+                'Article Attribute'
+            ],
+            'Variant Attribute' => [
```

Comment:
Something seems odd about this data provider: Shouldn't this data set actually fail because for Shopware 5.5 we try to get the attributes from the article not its main variation?

Reply:
Check why the test doesn't fail with this data set. It might be valid but we should still know the reason. Post your answer here.

Reply:
I checked the code, its because if there is an attribute on the root article, it assigns it in the mainDetail article attribute and then unsets the root article attribute index. So basically it is getting the attribute data from the main detail as the root article attribute is always null.
Thats just my observation. If you have some insight do let me know too.

Reply:
Interesting....It works in production at least. Let me do some checks.